.github/workflows/maintenance.yml

```diff
@@ -0,0 +1,36 @@
+name: Set maintenance mode
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        required: true
+        type: choice
+        options:
+          - test
+          - production
```

Comment:
Could we have preproduction as an option here too. 

Our DR testing will be done on that environment.